### PR TITLE
Completions for mathematical tools

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
## Description

Completions for mathematical tools.

## TODOs:
- [x] octave
- [x] matlab

### CODE REVIEW

**Comment by EmilySeville7cfg** [comment]
Are tests failing due to #8509 fix?

**Comment by krobelus** [comment]
On Thu, Dec 02, 2021 at 01:48:58AM -0800, Emily Grace Seville wrote:
> Are tests failing due to #8509 fix?

yeah, fixed now

File: share/completions/matlab.fish [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Completion for: MATLAB R2021b
+
+function __matlab_check_no_desktop_nodesktop_opts
```
Comment (krobelus) [inline share/completions/matlab.fish:3]:
```suggestion
function __fish__matlab_check_no_desktop_nodesktop_opts
```

File: share/completions/octave.fish [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Completion for: GNU Octave 5.2.0
+
+complete --command octave --exclusive --arguments '(__fish_complete_suffix .m)'
+
+complete --command octave --short-option h --long-option help --description 'Show help'
+complete --command octave --short-option v --long-option version --description 'Show version'
+
+complete --command octave --long-option built-in-docstrings-file --require-parameter \
+  --description 'Use docs for built-ins'
```
Comment (krobelus) [inline share/completions/octave.fish:9]: please use short options for common flags like -d

File: share/completions/octave.fish [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Completion for: GNU Octave 5.2.0
+
```
Comment (krobelus) [inline share/completions/octave.fish:2]: $status is the implicit return value, you can drop this line

File: share/completions/matlab.fish [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  --description 'Run the JVM software without opening the desktop'
+complete --command matlab --old-option nojvm \
+  --description 'Start without the JVM software'
+
+# Display options
+complete --command matlab --old-option noFigureWindows \
+  --description 'Disable the display of figure windows'
+complete --command matlab --old-option nosplash \
+  --description 'Don\'t display the splash screen during startup'
+complete --command matlab --old-option nodisplay \
```
Comment (krobelus) [inline share/completions/matlab.fish:28]: please squash your commits

**Comment by zanchey** [comment]
I addressed the feedback and style issues, and rebased with:
To github.com:fish-shell/fish-shell.git
   229e315fc..36699d49d  master -> master

**Pull request closed without merging**